doc/nrf/app_bootloaders.rst

```diff
@@ -0,0 +1,74 @@
+.. _app_bootloaders:
+
+Bootloaders and Device Firmware Updates
+#######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+There are two types of bootloaders used by the |NCS|:
+
+* :ref:`Immutable first-stage bootloaders <immutable_bootloader>`
+* :ref:`Upgradable second-stage bootloaders <upgradable_bootloader>`
+
+These types can be combined for use in products that need a second-stage bootloader.
+A second-stage bootloader adds the capability of providing device firmware updates to itself and the application.
+This is useful when a bootloader performs additional complex operations, like using a software stack.
+
+The bootloaders support two types of updates:
+
+* Direct updates - where an in-place substitution of the image takes place.
+* Background updates - where the updated image is obtained and stored, but the update is completed later on.
+
+You can deliver the updated images to the device in two ways:
+
+* Wired - where updates are sent through a wired connection, like UART, or delivered connecting a flash device.
+* Over-the-air (OTA) where updates are sent through a wireless connection, like Bluetooth LE.
+
+Not every bootloader supported in the |NCS| can be used as either a first-stage or a second-stage one.
```

Comment:
I'm confused by this sentence.
In two stages solution each bootloader entity has its static place: B0 is always the 1st stage and sdk-mcuboot is always the 2nd stage.
In case one stage is desired B0 or sdk-mcuboot can be used stand-alone.

Reply:
This sentence is intended to state that some bootloaders can only be a specific stage, like you pointed out about b0. Then readers consult the bootloader table to see which bootloader can be used in which stage.

This keeps the wording general in case more bootloaders than b0 or MCUboot become available in Zephyr/NCS. The table is the list of options and usability.

Perhaps this should just be removed, since the next sentence points to the table for bootloader info?

Reply:
Yes, the table is right after so maybe it is OK. I had written this comment before I looked ad the table which made this clear. I'm ok If you either keep  or modify this.